src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -43,18 +44,17 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
                         nameof(actionContext));
             }
 
+
+            var methodParameters = actionDescriptor.MethodInfo.GetParameters();
             var parameterMetadata = new List<ModelMetadata>();
             foreach (var parameter in actionDescriptor.Parameters)
             {
+                var parameterInfo = methodParameters.Where(p => p.Name == parameter.Name).Single();
                 var metadata = _modelMetadataProvider.GetMetadataForParameter(
-                    methodInfo: actionDescriptor.MethodInfo,
```

Comment:
> I still want to eliminate GetMetadataForParameter asap.
Yeah that would be ideal.

My problem was just that the api surface seemed in consistent. ( Having a GetMetadataForProperties and GetParameters would be consistent IMO)..